src/app/calculator/furnaces/flue-gas/energy-form/energy-form.component.ts

```diff
@@ -0,0 +1,128 @@
+  styleUrls: ['./energy-form.component.css']
+})
+export class EnergyFormComponent implements OnInit {
+  showOperatingHoursModal: boolean;
+  energyForm: FormGroup;
+  @Input()
+  settings: Settings;
+  @Input()
+  isBaseline: boolean;
+  @Input()
+  selected: boolean;
+  @ViewChild('formElement', { static: false }) formElement: ElementRef;
+
+  resetDataSub: Subscription;
+  generateExampleSub: Subscription;
+  showFlueGasModal: boolean;
+
+  formWidth: number;
+  energyUnit: string;
+  energySourceTypeSub: any;
+  constructor(private flueGasService: FlueGasService,
+             private cd: ChangeDetectorRef,
+             private energyFormService: EnergyFormService) { }
+
+  ngOnInit(): void {
+    this.initSubscriptions();
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes.selected && !changes.selected.firstChange) {
+      this.setFormState();
+    }
+  }
+
+  ngOnDestroy() {
+    this.resetDataSub.unsubscribe();
+    this.generateExampleSub.unsubscribe();
+  }
+
+  initSubscriptions() {
+    this.resetDataSub = this.flueGasService.resetData.subscribe(value => {
+      this.initForm();
+      });
+    this.generateExampleSub = this.flueGasService.generateExample.subscribe(value => {
+      this.initForm();
+    });
+  }
+
+  setFormState() {
+    if (this.selected == false) {
+      this.energyForm.disable();
+    } else {
+      this.energyForm.enable();
+    }
+    this.cd.detectChanges();
+  }
+
+
+  initForm() {
+    let energyData: EnergyData;
+    if (this.isBaseline) {
+      energyData = this.flueGasService.baselineEnergyData.getValue();
+    } else {
+      energyData = this.flueGasService.modificationEnergyData.getValue();
+    }
+    if (energyData) {
+      this.energyForm = this.energyFormService.getEnergyForm(energyData);
+    } else {
+      this.energyForm = this.energyFormService.initEnergyForm();
+    }
+
+    if (!this.energyForm.controls.fuelCost.value) {
+      this.energyForm.patchValue({
+        fuelCost: this.settings.fuelCost,
+      });
+    }
+
+    this.calculate();
+    this.setFormState();
+  }
+
+
+  focusField(str: string) {
+    this.flueGasService.currentField.next(str);
+  }
+
+  calculate() {
+    let currentEnergyData: EnergyData = this.energyFormService.buildEnergyData(this.energyForm);
+    if (this.isBaseline) {
+      this.flueGasService.baselineEnergyData.next(currentEnergyData);
+    } else {
+      this.flueGasService.modificationEnergyData.next(currentEnergyData);
+    }
+  }
+
+  closeOperatingHoursModal() {
+    this.showOperatingHoursModal = false;
+  }
+
+  openOperatingHoursModal() {
+    this.showOperatingHoursModal = true;
+  }
+
+  updateOperatingHours(oppHours: OperatingHours) {
+    this.flueGasService.operatingHours = oppHours;
+    this.energyForm.controls.hoursPerYear.patchValue(oppHours.hoursPerYear);
+    this.calculate();
+    this.closeOperatingHoursModal();
+  }
+
+  setOpHoursModalWidth() {
+    if (this.formElement.nativeElement.clientWidth) {
```

Comment:
just check `this.formElement`, it's a check to see if the object exists before accessing its properties